lsp-mode.el

```diff
@@ -7109,15 +7109,13 @@ reported according to `flycheck-check-syntax-automatically'."
   "Start an LSP syntax check with CHECKER.
 
 CALLBACK is the status callback passed by Flycheck."
-  (->> (or (gethash (lsp--fix-path-casing buffer-file-name)
-                    (lsp-diagnostics))
-           (gethash (lsp--fix-path-casing (file-truename buffer-file-name))
-                    (lsp-diagnostics)))
-       (-map (-lambda (diag)
+  (cl-labels ((make-flycheck-error
+               (file diag)
+               ;; when FILE is nil, use the current buffer
                (flycheck-error-new
-                :buffer (current-buffer)
+                :buffer (if file (find-file-noselect file) (current-buffer))
```

Comment:
That would mean we open all files for which we have received diagnostics which is no-go for performance reasons(imagine a legacy project with 1000+ files with warnings).

Reply:
Oh, I have been too long in Haskell land, I was thinking it would be a thunk or a monadic action (deferred computation in this case) :slightly_smiling_face:.

This function would only be called for *errors* in other buffers, not warnings.

Yeah, I wrote this with Haskell in mind, where the compiler typically stops when it encounters an error and doesn't move on to the next file. So I'd never run the risk of getting tons of errors in other buffers. In my workflow, you fix errors one by one, buffer per buffer, starting from the top of the module dependency graph. So you nearly always would want to switch to the other buffer containing errors.

Reply:
I had similar issues with Java and I created lsp-treemacs-errors-list. The general issue is that both flycheck and flymake are designed after single file liters and they don't have the notion of a project which does not fit in lsp world.